### PR TITLE
Docs 2784: Explain components, modules, connectors in one topic

### PR DESCRIPTION
**Pull request by rddheath**
Is this looking like what you believe is expected by the people who made the request?

### CODE REVIEW

File: mule4-user-guide/v/4.1/mule-application-about.adoc [hunk 1]
```diff
@@ -5,7 +5,16 @@ You typically create Mule apps to perform system integrations. Mule apps are con
 The main building blocks of Mule apps are connectors, components, and modules that provide connectivity to external servers, handle data transformations, and route and process Mule events.
 
 Components::
-Components execute business logic on the messages that flow through your apps.
```
Comment (Mulesoft-Mysti) [inline mule4-user-guide/v/4.1/mule-application-about.adoc]: Please provide one concrete example of a component, connector, and module so it's super clear. Just the name is fine.

File: mule4-user-guide/v/4.1/mule-application-about.adoc [hunk 2]
```diff
@@ -5,7 +5,16 @@ You typically create Mule apps to perform system integrations. Mule apps are con
...
+* Asynchronous processing of parts of a flow or subflow
+* Batch processing of messages
+* Caching and reusing frequently requested data
+* Initiating subflows
+* Logging
+* Setting payloads
+* Transforming messages with the DataWeave language
+* Creating Try scopes in your flows, so that you can catch and respond to errors
 +
 For more information about components, see "About Components", which is linked to from the *See Also* section at the end of this page.
```
Comment (Mulesoft-Mysti) [inline mule4-user-guide/v/4.1/mule-application-about.adoc]: Delete line 19, and the other two pointers to the See Also section. This defeats the purpose of a See Also section.

File: mule4-user-guide/v/4.1/mule-application-about.adoc [hunk 3]
```diff
@@ -4,7 +4,32 @@ You typically create Mule apps to perform system integrations. Mule apps are con
...
+* Caching and reusing frequently requested data
+* Initiating subflows
+* Logging
+* Setting payloads
+* Transforming messages with the DataWeave language
+* Creating Try scopes in your flows, so that you can catch and respond to errors
++
+For example, the Choice router is a component that dynamically routes messages through a flow according to a set of DataWeave expressions that evaluate message content. Each expression is associated with a different routing option. The effect is to add conditional processing to a flow, similar to an `if`/`then`/`else` code block in most programming languages.
+
+Connectors::
```
Comment (pablolagreca) [inline mule4-user-guide/v/4.1/mule-application-about.adoc]: We are putting modules and connectors at the same level of components and that's not accurate.

"Connectors" do not execute business logic on messages. Connectors and modules are a mechanism to group a set of related components. Connector group together a set of components that let you integrate applications with external systems (your explanation is perfect so use that. And Modules group together a set of components that perform related functionality like "aggregate values, compress data, use Java features, use extra features for processing JSON, and more."

File: mule4-user-guide/v/4.1/mule-application-about.adoc [hunk 4]
```diff
@@ -4,7 +4,32 @@ You typically create Mule apps to perform system integrations. Mule apps are con
...
++
+For example, the Salesforce Connector lets you connect to the Salesforce platform APIs. This connector exposes methods for accessing Salesforce, including working with the Salesforce Apex classes. This connector works with the Salesforce SOAP API, REST API, Bulk API, and Streaming API, depending on the operation you configure. Each API call uses an XML request and response over an HTTPS connection. All required request headers, error handling, and HTTPS connection configurations are built into the connector.
+
+Modules::
+Modules process messages, but they do not execute any business logic as they do so. Instead, they add flexibility to your apps by allowing you to aggregate values, compress data, use Java features, use extra features for processing JSON, and more. Modules are similar to components, but the features they offer are more at the level of traditional application coding.
++
+For example, the Scripting module can execute custom logic written in a scripting language. Supported scripting languages are Groovy, JavaScript, Python, and Ruby.
```
Comment (pablolagreca) [inline mule4-user-guide/v/4.1/mule-application-about.adoc]: this "the Scripting module can execute" would be "the Scripting module provides a component that can execute"

File: mule4-user-guide/v/4.1/mule-application-about.adoc (hunk 3 above)
Comment (rddheath) [reply mule4-user-guide/v/4.1/mule-application-about.adoc]: Are you speaking strictly from a developer's point of view or from a user's point of view? Connectors might group a set of related components, but to a user a connector looks like one object, right? The same goes for modules: modules group together a set of components, but the user sees one object, right? I think I should describe these things from the point of view of a user.

**Comment by rddheath** [comment]
@Mulesoft-Mysti , would you have another look?

**Pull request merged**